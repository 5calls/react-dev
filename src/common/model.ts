
export interface Issue {
  id: string;
  name: string;
  reason: string;
  script: string;
  contacts?: Contact[];
  contactType?: string;
  categories: Category[];
  inactive: boolean;
  outcomeModels: Outcome[];
  link: string;
  linkTitle: string;
  slug: string;
}

export const DefaultIssue: Issue = {
  id: '',
  name: '',
  reason: '',
  script: '',
  categories: [],
  inactive: false,
  outcomeModels: [],
  link: '',
  linkTitle: '',
  slug: '',
};

export interface Outcome {
  label: string;
  status: string;
}

export interface Contact {
  id: string;
  name: string;
  phone: string;
  photoURL?: string;
  party: Party;
  state: string;
  reason: string;
  area?: string;
  field_offices?: FieldOffice[];
}

export interface VoterContact {
  id: string;
  name: string;
  city: string;
  phone: string;
}

export interface Category {
  name: string;
}

export class CategoryMap {
  category: Category;
  issues: Issue[];
}

export type Party = 'Democrat' | 'Republican' | 'Independent' | '';

// export const DefaultContact: Contact = {
//   name: '',
//   phone: '',
//   party: '',
//   state: '',
//   reason: ''
// };

export const DefaultContact: Contact = {} as Contact;

export interface FieldOffice {
  city: string;
  phone: string;
}

export interface UserStat {
  all: string[];
  contactedCount: number;
  voiceMailCount: number;
  unavailableCount: number;
}

export interface Group {
  id: string;
  name: string;
  description: string;
  totalCalls: number;
}

/**
 * Represents the place used to get location.
 * It may be one of three options:
 * 1. CACHED_ADDRESS - address stored in local storage
 * 2. BROWSER_GEOLOCATION - address obtained from the browser geolocation API
 * 3. IP_INFO - address obtained from ipinfo.io API
 */
export enum LocationFetchType {
  CACHED_ADDRESS = 'CACHED_ADDRESS', // 'address' in Choo version
  BROWSER_GEOLOCATION = 'BROWSER_GEOLOCATION', // 'browserGeolocation' in Choo version
  IP_INFO = 'IP_INFO' // 'ipAddress' in Choo version
}

/**
 * Encapsulates a location using latitude
 * and longitude. Undefined for either longitude
 * or latitude indicates that the geolocation has
 * has not been set.
 */
export interface GeolocationPosition {
  longitude: number | undefined;
  latitude: number | undefined;
}

/* 5 Calls API data */
export interface ApiData {
  splitDistrict: boolean;
  invalidAddress: boolean;
  normalizedLocation: string | undefined;
  divisions: string[];
  issues: Issue[];
}

export interface GroupIssues {
  splitDistrict: boolean;
  invalidAddress: boolean;
  normalizedLocation: string | undefined;
  divisions: string[];
  issues: Issue[];
}

export interface CountData {
  count: number; // total call count
}

/* Data from iponfo.io API */
export interface IpInfoData {
  ip: string;
  hostname: string;
  city: string;
  region: string; // state
  country: string;
  loc: string; // long, lat - used in issue lookup
  org: string; // internet service provider
  postal: string; // zip code
}

export enum LocationUiState {
  FETCHING_LOCATION = 'FETCHING_LOCATION',
  LOCATION_FOUND = 'LOCATION_FOUND',
  ENTERING_LOCATION = 'ENTERING_LOCATION',
  LOCATION_ERROR = 'LOCATION_ERROR'
}

export interface DonationGoal {
  goal: Donations; //
}

export interface Donations {
    count: number; // number of donors
    amount: number; // total collected (api===amount)
    total: number; // goal (api===total)
    kind: string; // denomincation (dollars)
}
