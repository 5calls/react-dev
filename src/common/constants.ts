
export const APP_NAME = '5 Calls';
export const APP_URL = 'https://5calls.org';
// export const APP_URL = 'http://localhost:8090';
export const API_URL = 'https://api.5calls.org/v1';
export const ISSUES_API_URL = `${APP_URL}/issues/?all=true&address=`;
export const REPORT_API_URL = `${APP_URL}/report`;
export const COUNTS_API_URL = `${API_URL}/counts`;
export const DONATIONS_API_URL = `${API_URL}/donations`;
export const GROUP_API_URL = `${API_URL}/groups`;
export const IP_INFO_URL = 'https://ipinfo.io/json';

export const zipCodeRegex: RegExp = /^\d{5}(-\d{4})?$/;

export const contact = {
  email: 'make5calls@gmail.com',
  github: 'https://github.com/5calls',
  twitter: 'https://twitter.com/make5calls',
  facebook: 'https://www.facebook.com/make5calls',
  apps: 'https://crgj.app.link/7R2bEB0R4F',
};
