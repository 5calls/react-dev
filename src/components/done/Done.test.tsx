import * as React from 'react';
import { shallow } from 'enzyme';
import { Done } from './index';
import { DefaultIssue } from '../../common/model';

test('Done component snapshot renders correctly', () => {
  const issue = Object.assign({}, DefaultIssue, {id: 3});
  const count = 10000;
  const component = shallow(
    <Done
      currentIssue={issue}
      totalCount={count}
    />
  );
  expect(component).toMatchSnapshot();
});
