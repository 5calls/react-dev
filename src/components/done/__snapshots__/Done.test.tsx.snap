// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Done component snapshot renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="call"
>
    <div
        className="call_complete"
    >
        <h2
            className="call__title"
        />
        <p
            className="call__text"
        />
        <Translate(Promotion)
            currentIssue={
                Object {
                    "categories": Array [],
                    "id": 3,
                    "inactive": false,
                    "link": "",
                    "linkTitle": "",
                    "name": "",
                    "outcomeModels": Array [
                      Object {
                        "label": "contact",
                        "status": "contact",
                      },
                    ],
                    "reason": "",
                    "script": "",
                  }
            }
            t={[Function]}
        />
        <p
            className="call__text"
        >
            <Trans
                i18nKey="callComplete.learnWhyCallingIsGreat"
                parent="span"
            >
                <Link
                    replace={false}
                    to="/about"
                >
                    about-link-text-stub
                </Link>
            </Trans>
        </p>
        <Translate(Component)
            t={[Function]}
            totalCount={10000}
        />
    </div>
</section>,
  "nodes": Array [
    <section
      className="call"
>
      <div
            className="call_complete"
      >
            <h2
                  className="call__title"
            />
            <p
                  className="call__text"
            />
            <Translate(Promotion)
                  currentIssue={
                        Object {
                              "categories": Array [],
                              "id": 3,
                              "inactive": false,
                              "link": "",
                              "linkTitle": "",
                              "name": "",
                              "outcomeModels": Array [
                                Object {
                                  "label": "contact",
                                  "status": "contact",
                                },
                              ],
                              "reason": "",
                              "script": "",
                            }
                  }
                  t={[Function]}
            />
            <p
                  className="call__text"
            >
                  <Trans
                        i18nKey="callComplete.learnWhyCallingIsGreat"
                        parent="span"
                  >
                        <Link
                              replace={false}
                              to="/about"
                        >
                              about-link-text-stub
                        </Link>
                  </Trans>
            </p>
            <Translate(Component)
                  t={[Function]}
                  totalCount={10000}
            />
      </div>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        currentIssue={
                Object {
                        "categories": Array [],
                        "id": 3,
                        "inactive": false,
                        "link": "",
                        "linkTitle": "",
                        "name": "",
                        "outcomeModels": Array [
                          Object {
                            "label": "contact",
                            "status": "contact",
                          },
                        ],
                        "reason": "",
                        "script": "",
                      }
        }
        t={[Function]}
        totalCount={10000}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "currentIssue": Object {
            "categories": Array [],
            "id": 3,
            "inactive": false,
            "link": "",
            "linkTitle": "",
            "name": "",
            "outcomeModels": Array [
              Object {
                "label": "contact",
                "status": "contact",
              },
            ],
            "reason": "",
            "script": "",
          },
          "t": [Function],
          "totalCount": 10000,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="call"
>
          <div
                    className="call_complete"
          >
                    <h2
                              className="call__title"
                    />
                    <p
                              className="call__text"
                    />
                    <Translate(Promotion)
                              currentIssue={
                                        Object {
                                                  "categories": Array [],
                                                  "id": 3,
                                                  "inactive": false,
                                                  "link": "",
                                                  "linkTitle": "",
                                                  "name": "",
                                                  "outcomeModels": Array [
                                                    Object {
                                                      "label": "contact",
                                                      "status": "contact",
                                                    },
                                                  ],
                                                  "reason": "",
                                                  "script": "",
                                                }
                              }
                              t={[Function]}
                    />
                    <p
                              className="call__text"
                    >
                              <Trans
                                        i18nKey="callComplete.learnWhyCallingIsGreat"
                                        parent="span"
                              >
                                        <Link
                                                  replace={false}
                                                  to="/about"
                                        >
                                                  about-link-text-stub
                                        </Link>
                              </Trans>
                    </p>
                    <Translate(Component)
                              t={[Function]}
                              totalCount={10000}
                    />
          </div>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="call"
>
          <div
                    className="call_complete"
          >
                    <h2
                              className="call__title"
                    />
                    <p
                              className="call__text"
                    />
                    <Translate(Promotion)
                              currentIssue={
                                        Object {
                                                  "categories": Array [],
                                                  "id": 3,
                                                  "inactive": false,
                                                  "link": "",
                                                  "linkTitle": "",
                                                  "name": "",
                                                  "outcomeModels": Array [
                                                    Object {
                                                      "label": "contact",
                                                      "status": "contact",
                                                    },
                                                  ],
                                                  "reason": "",
                                                  "script": "",
                                                }
                              }
                              t={[Function]}
                    />
                    <p
                              className="call__text"
                    >
                              <Trans
                                        i18nKey="callComplete.learnWhyCallingIsGreat"
                                        parent="span"
                              >
                                        <Link
                                                  replace={false}
                                                  to="/about"
                                        >
                                                  about-link-text-stub
                                        </Link>
                              </Trans>
                    </p>
                    <Translate(Component)
                              t={[Function]}
                              totalCount={10000}
                    />
          </div>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    currentIssue={
        Object {
            "categories": Array [],
            "id": 3,
            "inactive": false,
            "link": "",
            "linkTitle": "",
            "name": "",
            "outcomeModels": Array [
              Object {
                "label": "contact",
                "status": "contact",
              },
            ],
            "reason": "",
            "script": "",
          }
    }
    t={[Function]}
    totalCount={10000}
/>,
}
`;
